checkstyle/suppressions.xml

```diff
@@ -340,6 +340,14 @@
     <suppress checks="NPathComplexity"
               files="CoordinatorRuntime.java"/>
 
+    <!-- share coordinator -->
+    <suppress checks="MethodLength"
+              files="ShareCoordinatorShardTest.java"/>
+    <suppress checks="NPathComplexity"
+              files="ShareCoordinatorShard.java"/>
+    <suppress checks="CyclomaticComplexity"
+              files="ShareCoordinatorShard.java"/>
```

Comment:
nit: While it's fine to avoid suppressions but avoid if we have better way to handle the situation in code. For `MethodLength` you might want to have a method to generate `new TestAttributes` or something similar to avoid method length issue.

Reply:
The problem will still remain, since the test cases are curated to create various scenarios. We cannot generate them deterministically. If I add a new private method to return the tests, it will still five MethodLength issue.
Best I can do is to suppress warning for this specific method.

Reply:
Suppressing just that method makese sense.